sdk/core/AzureCore/Source/Pipeline/PipelineStage.swift

```diff
@@ -119,9 +128,15 @@ extension PipelineStage {
                         completionHandler(.success(response), httpResponse)
                     }
                 case let .failure(pipelineError):
-                    self.on(error: pipelineError) { error, handled in
+                    let pipelineResponse = PipelineResponse(
+                        request: request.httpRequest,
+                        response: httpResponse,
+                        logger: request.logger,
+                        context: request.context
+                    )
+                    self.on(error: pipelineError, pipelineResponse: pipelineResponse) { _, handled in
                         if !handled {
-                            completionHandler(.failure(error), httpResponse)
+                            completionHandler(.failure(pipelineError), nil)
```

Comment:
Should we not be passing in the response here rather than `nil`?

Reply:
Fixed.